lib/bundles/inspec-compliance/target.rb

```diff
@@ -13,7 +13,48 @@ module Compliance
   class Fetcher < Fetchers::Url
     name 'compliance'
     priority 500
-    def self.resolve(target) # rubocop:disable PerceivedComplexity, Metrics/CyclomaticComplexity, Metrics/AbcSize
+    attr_reader :upstream_sha256
+
+    def initialize(target, opts)
+      super(target, opts)
+      if target.is_a?(Hash) && target.key?(:url)
+        @target = target[:url]
+        @upstream_sha256 = target[:sha256]
+      elsif target.is_a?(String)
+        @target = target
+        @upstream_sha256 = ''
+      end
+    end
+
+    def sha256
+      upstream_sha256.empty? ? super : upstream_sha256
```

Comment:
After talking to @chris-rock it sounds like we need some logic here to only use the upstream_sha256 for lockfile versions > 1 in order to not break vendored profiles with a lockfile using the old sha256 method.

Reply:
This was addressed in another comment [here](https://github.com/inspec/inspec/pull/3221/files#r211676981)